### ISSUE TITLE
How do you get scan for the bridge ip?

### ISSUE THREAD

**Issue by leossmith** [dyvoid/hue-air#3]
Hi i am starting a new flex app and i am trying to figure out how you get the bridge ip in order to use it later in the app. 

Thank you

**Comment by dyvoid** [comment]
You can use the discoverBridgeThroughPortal method in HueConnector. It uses the N-UPnP method (described here: http://bit.ly/1RL4SAL). A thing to note about this is that it uses the online Hue Portal to get the IP address, so it requires an internet connection to work. If that doesn't work for your particular use case you might want to consider developing some of the other methods of that guide.

**Comment by leossmith** [comment]
That worked great. However when i try to create a new user it says that the ip is not specified. Do i need to parse the ip and then create a new object? Thanks.

**Comment by dyvoid** [comment]
Yes, for any calls to work, you will need to have set the IP address. Are you using the HueManager or using HueConnector directly?

For HueManager:
Pass it in through the connect method or use the ipAddress setter.

For HueConnector:
Pass it in through the constructor or use the ipAddress setter.

Now you can create a new user. Assuming your bridge is up to date, take note that when creating a user, the userName parameter has been deprecated (http://www.developers.meethue.com/documentation/configuration-api). Just pass in deviceType and then store the username that the bridge will return in the response.

Exactly like the IP address, you can then pass the userName through setters or connect/constructor methods. After that you should be good to go on all other calls.

**Comment by leossmith** [comment]
Hi @dyvoid, i am using the HueConnector directly with discoverBridgeThroughPortal. I have the latest bridget (the square one).

thank you for the username part. 

I hope this does the trick.

**Comment by leossmith** [comment]
I still get an error about the username.

[{"error":{"type":7,"address":"/username","description":"invalid value, null,, for parameter, username"}}]

i am using like con.createUser("my_hue_app#iphone peter");

**Comment by dyvoid** [comment]
Did you also test what happens when you DO pass in a username?

Anyway, unfortunately I can't test this myself, since my Hue bridge and bulbs are in storage (I recently moved). However, I've pushed the most likely fix in a seperate branch. Please pull the bug-3/username branch and let me know if this fixes your issue.

**Comment by leossmith** [comment]
Great, that did the trick. This works perfect now. Now i can go on developing the real logic of my app. Thank you so much for your help. Again, awesome job.

**Comment by dyvoid** [comment]
Great, glad that did the trick! I would recommend you look at the HueManager as well. It's a layer on top of the HueConnector that takes away some of the complexity and custom parsing. It allows you to communicate with your lights and groups through fully typed instances. Good luck!
